src/main.cpp

```diff
@@ -2837,6 +2844,9 @@ bool static ConnectTip(CValidationState& state, const CChainParams& chainparams,
     for(unsigned int i=0; i < pblock->vtx.size(); i++)
         txChanged.push_back(std::make_tuple(pblock->vtx[i], pindexNew, i));
 
+    // update mempool stats
+    CStats::DefaultStats()->addMempoolSample(mempool.size(), mempool.DynamicMemoryUsage(), mempool.GetMinFee(GetArg("-maxmempool", DEFAULT_MAX_MEMPOOL_SIZE) * 1000000).GetFeePerK());
```

Comment:
I like the current layering concept. IMO its preferable that the `stats.cpp` module has no knowledge over the `txmempool.h`'s classes, etc., I prefer to keep it as "dump" as possible.